### PR TITLE
Mount alloc and task dirs in rkt

### PR DESCRIPTION
**Pull request by achanda**
- Split out a function in a helper module
- Use that function to get alloc and task dirs in the rkt driver, and mount those

### CODE REVIEW

**Comment by achanda** [comment]
Rkt docs here https://coreos.com/rkt/docs/latest/subcommands/run.html#mounting-volumes

**Comment by achanda** [comment]
The tests run fine on my box while travis fails consistently. Any idea why?

File: client/driver/utils.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package driver
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/nomad/client/allocdir"
+	"github.com/hashicorp/nomad/nomad/structs"
+)
+
+func getBindDirs(alloc *allocdir.AllocDir, task *structs.Task) ([]string, error) {
```
Comment (dadgar) [inline client/driver/utils.go]: I would undo this, see other comment.

File: client/driver/rkt.go [hunk 2]
```diff
@@ -160,6 +156,20 @@ func (d *RktDriver) Start(ctx *ExecContext, task *structs.Task) (DriverHandle, e
 	// Append the run command.
 	cmdArgs = append(cmdArgs, "run", "--mds-register=false", img)
 
+	// Mount allc and task dirs
+	binds, err := getBindDirs(ctx.AllocDir, task)
+	if err != nil {
+		return nil, err
+	}
+	for _, bind := range binds {
+		s := strings.Split(bind, ":")
```
Comment (dadgar) [inline client/driver/rkt.go]: You aren't really using the output of the getBindDirs directly, you are parsing its output. So rather create a specialized method that returns what you are doing below. There doesn't have to be shared code between the rkt/docker driver as they are outputting different formats.

**Comment by achanda** [comment]
Let me redo this in another branch

**Pull request closed without merging**